File: Sample-Java/app/src/main/java/com/izettle/payments/android/java_example/MainActivity.java
package com.izettle.payments.android.java_example;

import static com.izettle.android.commons.ext.state.StateExtKt.toLiveData;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.izettle.payments.android.sdk.IZettleSDK;
import com.izettle.payments.android.sdk.User;

public class MainActivity extends AppCompatActivity {

    private TextView loginStateText;
    private Button loginButton;
    private Button logoutButton;
    private Button openCardReaderButton;
    private Button openPayPalQrcButton;

    @Override
    public void  onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        loginStateText = findViewById(R.id.login_state);
        loginButton = findViewById(R.id.login_btn);
        logoutButton = findViewById(R.id.logout_btn);
        openCardReaderButton = findViewById(R.id.open_card_reader_btn);
        openPayPalQrcButton = findViewById(R.id.open_paypal_btn);

        toLiveData(IZettleSDK.Instance.getUser().getState()).observe(this, state -> {
            onAuthStateChanged(state instanceof User.AuthState.LoggedIn);
        });

        loginButton.setOnClickListener( v -> {
            IZettleSDK.Instance.getUser().login(this, Color.WHITE);
        });

        logoutButton.setOnClickListener( v -> {
            IZettleSDK.Instance.getUser().logout();
        });

        openCardReaderButton.setOnClickListener( v -> {
            Intent intent = new Intent(this, CardReaderActivity.class);
            startActivity(intent);
        });

        openPayPalQrcButton.setOnClickListener( v -> {
            Intent intent = new Intent(this, PayPalQrcActivity.class);
            startActivity(intent);
        });
    }

    @SuppressLint("SetTextI18n")
    private void onAuthStateChanged(boolean isLoggedIn) {
        String state = "Unauthenticated";
        if(isLoggedIn) {
            state = "Authenticated";
        }
        loginStateText.setText("State: "+ state);
        loginButton.setEnabled(!isLoggedIn);
        logoutButton.setEnabled(isLoggedIn);
    }
}

File: Examples/Example-Kotlin/app/src/main/java/com/izettle/payments/android/kotlin_example/MainActivity.kt
package com.izettle.payments.android.kotlin_example

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.text.Editable
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.izettle.android.commons.ext.state.toLiveData
import com.izettle.payments.android.sdk.IZettleSDK
import com.izettle.payments.android.sdk.User.AuthState.LoggedIn

class MainActivity : AppCompatActivity() {

    private lateinit var loginStateText: TextView
    private lateinit var loginButton: Button
    private lateinit var logoutButton: Button
    private lateinit var openCardReaderButton: Button
    private lateinit var openPayPalQrcButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        loginStateText = findViewById(R.id.login_state)
        loginButton = findViewById(R.id.login_btn)
        logoutButton = findViewById(R.id.logout_btn)
        openCardReaderButton = findViewById(R.id.open_card_reader_btn)
        openPayPalQrcButton = findViewById(R.id.open_paypal_btn)

        IZettleSDK.user.state.toLiveData().observe(this) { state ->
            onAuthStateChanged(state is LoggedIn)
        }

        loginButton.setOnClickListener {
            IZettleSDK.user.login(this, Color.WHITE)
        }

        logoutButton.setOnClickListener {
            IZettleSDK.user.logout()
        }

        openCardReaderButton.setOnClickListener {
            val intent = Intent(this, CardReaderActivity::class.java)
            startActivity(intent)
        }

        openPayPalQrcButton.setOnClickListener {
            val intent = Intent(this, PayPalQrcActivity::class.java)
            startActivity(intent)
        }
    }

    @SuppressLint("SetTextI18n")
    private fun onAuthStateChanged(isLoggedIn: Boolean) {
        loginStateText.text = "State: ${if (isLoggedIn) "Authenticated" else "Unauthenticated"}"
        loginButton.isEnabled = !isLoggedIn
        logoutButton.isEnabled = isLoggedIn
    }
}

internal fun Editable.toLong(): Long? = try {
    val content = this.toString()
    content.toLong()
} catch (e: Exception) {
    null
}